File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=INT_MIN,cur=0;
        for(auto i:nums){
            if(cur<0)cur=0;
            cur+=i;
            sum=max(cur,sum);
        }
        return sum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        s,ma=0,float('-inf')
        for i in range(len(nums)):
            s+=nums[i]
            if s>ma:
                ma=s
            if s<0:
                s=0
        return ma

        